src/storage/cache/rpc/client.rs

```diff
@@ -0,0 +1,652 @@
+        }
+
+        // Set to packet task
+        let keep_alive_msg = ReqHeader {
+            seq: current_seq,
+            op: ReqType::KeepAliveRequest.to_u8(),
+            len: 0,
+        }
+        .encode();
+
+        if let Ok(()) = self.send_data(&keep_alive_msg).await {
+            debug!("{:?} Success to sent keep alive message", self);
+            Ok(())
+        } else {
+            debug!("{:?} Failed to send keep alive message", self);
+            Err(RpcError::InternalError(
+                "Failed to send keep alive message".to_owned(),
+            ))
+        }
+    }
+
+    /// Send packet by the client.
+    pub async fn send_packet(&self, req_packet: &mut P) -> Result<(), RpcError<String>> {
+        let current_seq = self.next_seq();
+        req_packet.set_seq(current_seq);
+        debug!("{:?} Try to send request: {:?}", self, current_seq);
+
+        // Send may be used in different threads, so we need to create local buffer to store the request data
+        if let Ok(req_buffer) = req_packet.get_req_data() {
+            let mut req_header = ReqHeader {
+                seq: req_packet.seq(),
+                op: req_packet.op(),
+                len: usize_to_u64(req_buffer.len()),
+            }
+            .encode();
+
+            // concate req_header and req_buffer
+            req_header.extend_from_slice(&req_buffer);
+
+            if let Ok(()) = self.send_data(&req_header).await {
+                debug!("{:?} Sent request success: {:?}", self, req_packet.seq());
+                // We have set a copy to keeper and manage the status for the packets keeper
+                // Set to packet task with clone
+                self.get_packets_keeper_mut().add_task(&mut req_packet.clone())?;
+
+                Ok(())
+            } else {
+                debug!("{:?} Failed to send request: {:?}", self, req_packet.seq());
+                Err(RpcError::InternalError("Failed to send request".to_owned()))
+            }
+        } else {
+            debug!(
+                "{:?} Failed to serialize request: {:?}",
+                self,
+                req_packet.seq()
+            );
+            Err(RpcError::InternalError(
+                "Failed to serialize request".to_owned(),
+            ))
+        }
+    }
+
+    /// Receive packet by the client
+    pub fn recv_packet(&self, resp_packet: &P) -> Option<P> {
+        // Try to receive the response from innner keeper
+        let packets_keeper: &mut PacketsKeeper<P> = self.get_packets_keeper_mut();
+        packets_keeper.consume_task(resp_packet.seq())
+    }
+
+    /// Receive loop for the client.
+    pub async fn recv_loop(&self) {
+        // Check and clean keeper timeout unused task
+        // The timeout data will be cleaned by the get function
+        // We don't need to clean the timeout data radically
+        let mut tickers = Box::pin(tokio::time::interval(
+            self.timeout_options.idle_timeout * 100,
```

Comment:
idle_timeout should be rename to task_timeout.

ticker should be task_timeout / 2 ,   if ticker > task_timeout,   the actual timeout will be inacurate

Reply:
idle_timeout to task_timeout

Reply:
Changed to `self.timeout_options.task_timeout.div_f32(2.0)`.